src/linear_algebra/alternating.lean

```diff
@@ -0,0 +1,222 @@
+/-
+Copyright (c) 2020 Zhangir Azerbayev. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Eric Wieser, Zhangir Azerbayev
+-/
+
+import linear_algebra.multilinear
+import group_theory.perm.sign
+
+/-!
+# Alternating Maps
+
+We construct the bundled function `alternating_map`, which extends `multilinear_map` with all the
+arguments of the same type.
+
+## Main definitions
+* `alternating_map R M N ι` is the space of `R`-linear alternating maps from `ι → M` to `N`.
+* `f.map_eq_args` expresses that `f` is zero when two inputs are equal.
+* `f.map_swap` expresses that `f` is negated when two inputs are swapped.
+* `f.map_perm` expresses how `f` varies by a sign change under a permutation of its inputs.
+* An `add_comm_monoid`, `add_comm_group`, and `semimodule` structure over `alternating_map`s that
+  matches the definitions over `multilinear_map`s.
+
+## Implementation notes
+`alternating_map` is defined in terms of `map_eq_args`, as this is easier to work with than
+using `map_swap` as a definition, and does not require `has_neg N`.
+-/
+
+-- semiring / add_comm_monoid
+variables {R : Type*} [semiring R]
+variables {M : Type*} [add_comm_monoid M] [semimodule R M]
+variables {N : Type*} [add_comm_monoid N] [semimodule R N]
+
+-- semiring / add_comm_group
+variables {L : Type*} [add_comm_group L] [semimodule R L]
+
+-- ring / add_comm_group
+variables {R' : Type*} [ring R']
+variables {M' : Type*} [add_comm_group M'] [semimodule R' M']
+variables {N' : Type*} [add_comm_group N'] [semimodule R' N']
+
+variables {ι : Type*} [decidable_eq ι]
+
+set_option old_structure_cmd true
+
+section
+variables (R M N ι)
+/--
+An alternating map is a multilinear map that vanishes when two of its arguments are equal.
+-/
+structure alternating_map extends multilinear_map R (λ i : ι, M) N :=
+(map_eq_args' : ∀ (v : ι → M) (i j : ι) (h : v i = v j) (hij : i ≠ j), to_fun v = 0)
+end
+
+attribute [nolint doc_blame] alternating_map.to_multilinear_map
```

Comment:
```suggestion
/-- The multilinear map associated to an alternating map -/
add_decl_doc alternating_map.to_multilinear_map
```
Isn't this better? Or not?

Reply:
I figured this wasn't that interesting, and we already ignore the linter on things like `linear_equiv.to_linear_map`, `alg_equiv.to_ring_equiv`, etc.

Reply:
It's true that all these maps are obvious, but for sake of completeness I think it'd be good to provide doc strings for all those nolints too eventually.

Reply:
The linter already doesn't complain about things like `comm_ring.to_comm_semiring`, or docstrings being missing on projections - it seems to me that this is a case of the linter complaining about something it shouldn't.

Reply:
I don't think that having these docstrings hurts. So let's just have them.